cli/admin/src/main/java/io/pravega/cli/admin/controller/ControllerCommand.java

```diff
@@ -47,14 +58,39 @@
      * @return REST client.
      */
     protected Context createContext() {
+        CLIControllerConfig config = getCLIControllerConfig();
         ClientConfig clientConfig = new ClientConfig();
         clientConfig.register(JacksonJsonProvider.class);
         clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
-        Client client = ClientBuilder.newClient(clientConfig);
+
+        Client client;
+
+        // If TLS parameters are configured, set them in client.
+        if (config.isTlsEnabled()) {
+            SSLContext tlsContext;
+            try {
+                KeyStore ks = createTrustStore(config.getTruststore());
+                TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                tmf.init(ks);
+                tlsContext = SSLContext.getInstance("TLS");
+                tlsContext.init(null, tmf.getTrustManagers(), null);
+            } catch (KeyStoreException | IOException | NoSuchAlgorithmException | CertificateException | KeyManagementException e) {
+                output("Encountered exception while trying to use the given truststore: %s", config.getTruststore());
+                e.printStackTrace();
+                return null;
+            }
+            client = ClientBuilder.newBuilder()
+                    .withConfig(clientConfig)
+                    .sslContext(tlsContext)
+                    .build();
+        } else {
+            client = ClientBuilder.newClient(clientConfig);
+        }
+
         // If authorization parameters are configured, set them in the client.
-        if (getCLIControllerConfig().isAuthEnabled()) {
-            HttpAuthenticationFeature auth = HttpAuthenticationFeature.basic(getCLIControllerConfig().getUserName(),
-                    getCLIControllerConfig().getPassword());
+        if (config.isAuthEnabled()) {
```

Comment:
nit: The logic here'll be easier to follow with a slight rearrangement. Something like:

```
builder = ClientBuilder.newBuilder().withConfig(clientConfig); 
if (tls enabled) {
     builder.sslContext(...)
} 
Client client = builder.build(); 

if (auth enabled) {
     // do auth-specific stuff
}
Client client = builder.build(); 

```

Reply:
Following a pattern like this